src/main/java/org/folio/circulation/domain/LastCheckIn.java

```diff
@@ -0,0 +1,50 @@
+
+package org.folio.circulation.domain;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.folio.circulation.support.JsonPropertyWriter.write;
+
+import java.lang.invoke.MethodHandles;
+import java.util.UUID;
+
+import io.vertx.core.json.JsonObject;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Information about when an item was last scanned in the Inventory app.
+ *
+ */
+public class LastCheckIn {
+
+  private static final Logger log = LoggerFactory.getLogger(
+    MethodHandles.lookup().lookupClass());
+
+  private final DateTime dateTime;
+  private final UUID servicePointId;
+  private final String staffMemberId;
+
+  public LastCheckIn(DateTime dateTime, UUID servicePointId, String staffMemberId) {
+    this.dateTime = dateTime;
+    this.servicePointId = servicePointId;
+    this.staffMemberId = resolveStaffMemberId(staffMemberId);
+  }
+
+  private String resolveStaffMemberId(String id){
+    try {
+      return UUID.fromString(id).toString();
```

Comment:
What is the purpose of parsing the string into a UUID and back?

Reply:
To verify for empty, null, or invalid uuid

Reply:
Basically this covers all the scenarious when the herder, where the useId comes from, does not provide it. Since this is the ubiquitous system-wide header, chances this is going to happen are close to 0. Can we rely on the heade being set always, can't we?

Reply:
> Basically this covers all the scenarios when the header, where the useId comes from, does not provide it. Since this is the ubiquitous system-wide header, chances this is going to happen are close to 0. Can we rely on the heade being always set, can't we?

The header is provided by a client to the module, I don’t think we can rely on it being present. This is why I asked about the requirements, because I think the system should react when it isn’t present. 

Personally, I would fail the process if it is not present. However, one of your previous comments stated that the provided requirements said it could just be omitted in those cases.

What I don’t understand, if why parse it at all? Are we assuming that if it is a UUID then it is valid otherwise it is not?

Reply:
UUID parsing has been removed.